src/com/seattletechinterviews/datastructures/MyLinkedList.java

```diff
@@ -1,27 +1,92 @@
 package com.seattletechinterviews.datastructures;
 
 public interface MyLinkedList {
-	/**
+
+	 class Node {
+	 	int value;
+	 	Node next;
+	 	
+	 	public Node (int value){
+	 		this.value = value;
+	 	}
+	 }
+	 /**
 	 * Add a value to the head of the linked list.
 	 * @param value
 	 */
-	public void add(int value);
+	public void add(Node head, int value){
+	
+		head.value = value;
+	};
+	
 	/**
 	 * Remove the node at a particular location and return the 
 	 * value at that node. The node location start at 0 and 0 is 
 	 * the head. 
 	 * @param location
 	 */
-	public int removeAt(int location);
+	public int removeAt(Node head, int location){
+		Node previous = head;
+		int headMoved = 0;
+		if (head == null){
+			return null;
+		}// if location is the first node
+		else if (head.value == location){
+			return head.value;
+		}
+		// if location is in the middle
+		while (head.next != null){
+			head = head.next;
+			headMoved++;
+			if (headMoved >= 2){
+				previous = previous.next;
+			}
+			if (head.value == location){
```

Comment:
The ask is not to search for the value - the ask is to remove the node at the given location and return the value at the location.